vm.go

```diff
@@ -525,6 +528,74 @@ func (vm *VirtualMachineSpec) UpdateGpuConfiguration(mode GpuAssignmentMode, all
 	return nil
 }
 
+// Add vPCI device
+func (vm *VirtualMachineSpec) AssignDevice(ctx context.Context, deviceID string) (string, error) {
+	system, err := hcs.OpenComputeSystem(ctx, vm.ID)
+	if err != nil {
+		return "", err
+	}
+	defer system.Close()
+
+	guid, err := guid.NewV4()
+	if err != nil {
+		return "", err
+	}
+
+	vmBusGUID := guid.String()
+	targetDevice := hcsschema.VirtualPciDevice{
+		Functions: []hcsschema.VirtualPciFunction{
+			{
+				DeviceInstancePath: deviceID,
```

Comment:
>deviceID [](http://example.com/codeflow?start=24&length=8)

what happens if the device id already added to the compute system?

Reply:
will need to check that behavior on real hardware... will update once i find out...